### ISSUE TITLE
MRN Mac

### ISSUE THREAD

**Issue by klfaustbe** [Z-H-Sun/CS_CCME_Posts#3]
Hi All, hi @Z-H-Sun,

The patcher for MRN mac doesn't seem to work anymore. I tried on the 14.1.2 that was working before and on the new 14.2. After patching using a sudo command, the software starts and then closes immediately... Not sure why!

Any idea how to solve that issue? 

Thanks a lot!

**Comment by Z-H-Sun** [comment]
Well it is probable that the patch has been intentionally blocked by the developer. C’est la vie. But without running a test I cannot identify what happens. Unfortunately I have been quite busy recently. So I would suggest that you revert to 14.1 for now. I will keep you posted once I get a chance to work on it

**Comment by klfaustbe** [comment]
Thanks a lot! I did try to revert to 14.1 and now get the same issue... maybe because I upgraded my iOS to the last one? I get a “kill:9” issue when I try to launch the software through the terminal.

Anyway, I’d be more than grateful for any help :)

________________________________
De : CS_CCME <notifications@github.com>
Envoyé : Sunday, November 15, 2020 10:46:36 PM
À : Z-H-Sun/CS_CCME_Posts <CS_CCME_Posts@noreply.github.com>
Cc : klfaustbe <kl9361u@gmail.com>; Author <author@noreply.github.com>
Objet : Re: [Z-H-Sun/CS_CCME_Posts] MRN Mac (#3)


Well it is probable that the patch has been intentionally blocked by the developer. C’est la vie. But without running a test I cannot identify what happens. Unfortunately I have been quite busy recently. So I would suggest that you revert to 14.1 for now. I will keep you posted once I get a chance to work on it

**Comment by Z-H-Sun** [comment]
Ok, I might know what is happening. Please uninstall and reinstall M Nova, first of all. DO NOT use the patcher! Run M Nova once and see if it can be opened (at this time, it is not activated yet, but it is okay and we need this step). THEN, try running the patcher followed by M Nova.

This is because MacOS will check the digital signature of the app. After modifying the binary executable, the signature will be of course corrupted. If you do not allow macOS to run the check before patching , it will block the “untrusted” app.

Let me know if this works out. If not , please attach a screenshot of the patcher

**Comment by klfaustbe** [comment]
Hi,

I did actually run M nova alone first.

So here is what is happening now :

With 14.2 and 14.1.2 – once M nova is patched, the red opening sign comes up then immediately closes.
With 14.1.0 and sometimes 14.1.2 too, patched M Nova opens but then seems to freeze, I can move the windows around but I can’t click on anything – it’s irresponsive. Then if I try to reopen it it doesn’t work anymore.

Again, a huge thank for your help, this is really helpful in these stressful times!








Ok, I might know what is happening. Please uninstall and reinstall M Nova, first of all. DO NOT use the patcher! Run M Nova once and see if it can be opened (at this time, it is not activated yet, but it is okay and we need this step). THEN, try running the patcher followed by M Nova.

This is because MacOS will check the digital signature of the app. After modifying the binary executable, the signature will be of course corrupted. If you do not allow macOS to run the check before patching , it will block the “untrusted” app.

Let me know if this works out. If not , please attach a screenshot of the patcher

**Comment by klfaustbe** [comment]
I’m not sure if that helps,

But if I try to run the MRN (patched) through the terminal I get the following messages.

For 14.1.0
[cid:image001.png@01D6BBEA.9D4C8F30]

For 14.1.2 I get a “killed 9” what ever that is
[cid:image002.png@01D6BBEA.9D4C8F30]
Again, thank you!





Ok, I might know what is happening. Please uninstall and reinstall M Nova, first of all. DO NOT use the patcher! Run M Nova once and see if it can be opened (at this time, it is not activated yet, but it is okay and we need this step). THEN, try running the patcher followed by M Nova.

This is because MacOS will check the digital signature of the app. After modifying the binary executable, the signature will be of course corrupted. If you do not allow macOS to run the check before patching , it will block the “untrusted” app.

Let me know if this works out. If not , please attach a screenshot of the patcher

**Comment by Z-H-Sun** [comment]
I finally had a chance to run a quick check on the new versions of M Nova. The logic of verification process has changed in recent versions, unfortunately. The latest version that can be patched is **14.1.1** (tested on my macOS 10.14). Please try 14.1.1 and LMK if it works.
![image](https://user-images.githubusercontent.com/48094808/99867271-156cdc80-2b86-11eb-920f-6814231c8369.png)

I can't guarantee of coming up with a workable patch for 14.2 in the near future. Maybe it would be simple. Who knows?

**Comment by Z-H-Sun** [comment]
It is also possible that the M Nova team also modified the contents of previous version. So I am attaching a copy of 14.1.1 that I downloaded earlier. It is this version that I tested successfully with the patch. This copy is authentic and is not modified (not patched), and you can verify its digital signature. You can then try if the patch is successful.

https://drive.google.com/file/d/1EjpmGG6G2W7XJ2kLO_u1UMGSyugIEBmw/view?usp=sharing

**Comment by Z-H-Sun** [comment]
Also FYI, direct reply to the email does not include attachments such as images correctly. If you want to show me some pictures, please comment here on GitHub.

**Comment by klfaustbe** [comment]
Thanks a lot for your help! I really appreciate it! Let’s hope that the V14.2 will be patchable :)

________________________________
De : CS_CCME <notifications@github.com>
Envoyé : Saturday, November 21, 2020 4:34:54 AM
À : Z-H-Sun/CS_CCME_Posts <CS_CCME_Posts@noreply.github.com>
Cc : klfaustbe <kl9361u@gmail.com>; Author <author@noreply.github.com>
Objet : Re: [Z-H-Sun/CS_CCME_Posts] MRN Mac (#3)


Also FYI, direct reply to the email does not include attachments such as images correctly. If you want to show me some pictures, please comment here on GitHub.

**Comment by klfaustbe** [comment]
Sooooo I tried it on 14.1 and M Nova does open this time! However, after a few seconds, if you try to do anything (open a file, draw a structure, etc.) it freezes completely and I have to kill the process to get out.

**Comment by Z-H-Sun** [comment]
That is weird. I have not been met with such issues on my end. I think maybe we can do the following: We exchange our patched `Mest Re Nova` executable (Show package contents -> Contents -> MacOS -> Mest Re Nova), and find 1) if they have difference 2) what will happen if replacing our own with each other's (you may need to `chmod +x <path/to/>Mest Re Nova` after downloading).

Mine (patched) is shared through this link: https://drive.google.com/file/d/16uPDLkEpsLBelv3Nhi6n4Hk4yp5pquxt/view?usp=sharing

**Comment by klfaustbe** [comment]
That's strnage! I have two macs and both have the same issue... Also they both had the 14.2 patched with the "old" patch from your website and it was working even though the patch showed an error when run.

I wonder if this is because both mac have been upgraded to the new iOS... I'm attaching my file here.
[Mest Re Nova.zip](https://github.com/Z-H-Sun/CS_CCME_Posts/files/5578054/MestReNova.zip)

**Comment by klfaustbe** [comment]
I do experience the same freeze with your file... I think this must be an issue with MacOs Big Sure.

When I sudo it, I get this error:

QObject::connect: No such slot TUvirPluginGui::onClipboardDataChanged()
QObject::connect:  (receiver name: 'elvis')

**Comment by Z-H-Sun** [comment]
I checked the hash values and your file is exactly the same as mine. That means the patch is working properly. It is possible that older versions of M Nova is not compatible with the latest macOS. The "error messages" you mentioned seems not a huge issue: I had those as well, and they are more of warning messages from `QT` instead of errors.

To quickly check the compatibility, please try running the unpatched version:
```sh
cd /Applications/Mest Re Nova.app/Contents/MacOS
chmod +x Mest Re Nova.bak
./Mest Re Nova.bak
```

Now, it is not activated. Ignore verification errors, try drawing a molecule (should work like activated version) and opening an NMR spectrum file (should prompt that you do not have activated NMR plugin), and see if the window freezes.

**Comment by klfaustbe** [comment]
It's working fine :) So the issue seems to be between the patched version and MacOs Big Sure... We are narrowing it down :) Again thanks for your help!

**Comment by Z-H-Sun** [comment]
How about replacing the broken digital signature?

`sudo codesign --force --deep --sign - /Applications/Mest Re Nova.app`

You need to have XCode installed first of all.

**Comment by klfaustbe** [comment]
Let me download that and try! :)

**Comment by Z-H-Sun** [comment]
To install xcode, try running the following in the terminal:

`xcode-select --install`

**Comment by klfaustbe** [comment]
Same issue.

It's strange because I could use the top menu (although not everything works, I can't open a file using the open option, but I can open a file that is in the "recently opened" option)
![image](https://user-images.githubusercontent.com/74511108/99884377-c4142a00-2c25-11eb-8f8b-06ed76e1a582.png)

However if I try to click on the tab in the software itself, nothing is responding to my clicks. I can't even switch tabs. And as I said, I have the exact same issue on two different macs.

![image](https://user-images.githubusercontent.com/74511108/99884388-db531780-2c25-11eb-8064-76b9360a329e.png)

**Comment by klfaustbe** [comment]
Could this be the issue?

rocess:               nmrpredict-server [1297]
Path:                  /Applications/Mest Re Nova.app/Contents/Resources/NMRPredict/nmrpredict-server
Identifier:            nmrpredict-server
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        Mest Re Nova [1290]
Responsible:           Mest Re Nova [1290]
User ID:               501

Date/Time:             2020-11-21 18:13:06.624 +0000
OS Version:            macOS 11.0.1 (20B29)
Report Version:        12
Anonymous UUID:        D1CE56AE-15DA-B298-E2FD-C1ACCBFE4CDD

Sleep/Wake UUID:       D8147604-C4F7-40C4-975B-46FC314181D5

Time Awake Since Boot: 1200 seconds
Time Since Wake:       320 seconds

System Integrity Protection: disabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x1] Library missing

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  dyld: No shared cache present
Library not loaded: /usr/lib/libSystem.B.dylib
  Referenced from: /Applications/Mest Re Nova.app/Contents/Resources/NMRPredict/nmrpredict-server
  Reason: image not found

Binary Images:
       0x100c6b000 -        0x101880fff +nmrpredict-server (???) <C965E9EF-8E4F-3ED9-AB64-4EC8E2335B60> /Applications/Mest Re Nova.app/Contents/Resources/NMRPredict/nmrpredict-server
       0x11a6d4000 -        0x11a7f2fff +libgfortran.3.dylib (0) <39851E49-D973-36C6-9B95-9004CE8498B3> /Applications/Mest Re Nova.app/Contents/Resources/NMRPredict/libgfortran.3.dylib
       0x11a866000 -        0x11a89dfff +libquadmath.0.dylib (0) <A103C3CC-701F-3FAD-8C7C-387FBB5B1D2C> /Applications/Mest Re Nova.app/Contents/Resources/NMRPredict/libquadmath.0.dylib
    0x7fff5eeb2000 -     0x7fff5ef4dfff  dyld (832.7.1) <2705F0D8-C104-3DE9-BEB5-B1EF6E28656D> /usr/lib/dyld

**Comment by klfaustbe** [comment]
I'm attaching the different log files I got.


[Archive.zip](https://github.com/Z-H-Sun/CS_CCME_Posts/files/5578155/Archive.zip)

**Comment by Z-H-Sun** [comment]
That is very likely the reason. When it is not activated, the plugin won't run, so the issue was not revealed until you activated the app. (That being said, that issue is indeed due to compatibility issues between M Nova and the latest macOS instead of the patch.) However, there is little I could help, because I actually do not own a physical macbook, and I cannot debug on the latest macOS. I might try my friend's mac but it will take a while. Thanks for the detailed info.

**Comment by klfaustbe** [comment]
So I think I have found the issue:

https://www.techradar.com/how-to/macos-11-big-sur-problems-how-to-fix-the-most-common-issues

They moved the libraries to another location on the new MacOS - I guess one would have to make that change on the patch but I don't know how to do it sadly.

**Comment by klfaustbe** [comment]
Sorry, here is the link:

https://news.ycombinator.com/item?id=23612772

**Comment by klfaustbe** [comment]
This could help to:

https://github.com/urllib3/urllib3/pull/1905

**Comment by Z-H-Sun** [comment]
Awesome. We are getting close. It will take a while for me to find a solution. The temporary workarond, I think, is to disable the nmrpredict plugin. Can you delete the corresponding license file(s), `NMRPredictDesktop.lic` and `Random Forest NMR Predictor.lic`, and try if it works or if it produces new error log entries?

**Comment by klfaustbe** [comment]
bingo... it works without these licences!

**Comment by Z-H-Sun** [comment]
okie dokie. I actually dont find the prediction function very useful. Anyways, let me find if there is any way to make the predictor functionable.

**Comment by klfaustbe** [comment]
Ok best news ever:

If you install the latest version 14.2, open it, close it, patch it and then fix the signature with sudo codesign --force --deep --sign - /Applications/Mest Re Nova.app... IT WORKS! including the predictor, etc.

**Comment by Z-H-Sun** [comment]
Good. I assume that to solve the compatibility issues, they rewrote the code a little bit, so the assembly code I saw was changed, but the patch somehow still works. I will update the post and close the issue. Thank you a lot for testing and sharing.

**Comment by klfaustbe** [comment]
That’s cool! Thanks for helping!

**Comment by xgsdhm** [comment]
Hello, AFTER I run the MestReNovaCrack patch on the Window computer, it shows an error and cannot be cracked. I would like to ask your method. The version I installed is "mestrenova-12.0.4-22023_x64". The error code is as follows:

Processing D:\MNN\MestReNova.exe...
This file might have already been patched [Press `Y' to override].
Found pattern, JNZ -> JMP: 75 .. .* 0B 00 00 80 .* 0B 00 00 80
Error occurred:
#<Errno::EBADF: Bad file descriptor - D:\MNN\MestReNova.exe>
["MestReNovaCrack.rb:34:in `seek'", "MestReNovaCrack.rb:34:in `patch'", "MestReNovaCrack.rb:19:in `loop'", "MestReNovaCrack.rb:19:in `patch'", "MestReNovaCrack.rb:103", "MestReNovaCrack.rb:101:in `each'", "MestReNovaCrack.rb:101"]

Releasing .LIC files in D:\MNN\licenses...
All .LIC files have been created.

**Comment by Z-H-Sun** [comment]
> Hello, AFTER I run the Mest Re NovaCrack patch on the Window computer, it shows an error and cannot be cracked. I would like to ask your method. The version I installed is "mest re nova-12.0.4-22023_x64". The error code is as follows:
> 
> Processing D:\MNN\Mest Re Nova.exe... This file might have already been patched [Press `Y' to override]. Found pattern, JNZ -> JMP: 75 .. .* 0B 00 00 80 .* 0B 00 00 80 Error occurred: #<Errno::EBADF: Bad file descriptor - D:\MNN\Mest Re Nova.exe> ["Mest Re NovaCrack.rb:34:in `seek'", "Mest Re NovaCrack.rb:34:in `patch'", "Mest Re NovaCrack.rb:19:in `loop'", "Mest Re NovaCrack.rb:19:in `patch'", "Mest Re NovaCrack.rb:103", "Mest Re NovaCrack.rb:101:in `each'", "Mest Re NovaCrack.rb:101"]
> 
> Releasing .LIC files in D:\MNN\licenses... All .LIC files have been created.

plz do not leave irrelevant comments under this issue. all replies will be addressed in #8
